meinberlin/apps/budgeting/assets/Pagination.jsx

```diff
@@ -1,31 +1,37 @@
 import React from 'react'
 import django from 'django'
 import { PaginationButton } from './PaginationButton'
+import { useSearchParams } from 'react-router-dom'
 
 const pageNavigationStr = django.gettext('Page navigation')
 const pageNextStr = django.gettext('next page')
 const pagePrevStr = django.gettext('prev page')
 
 export const Pagination = (props) => {
   const {
-    currentPage,
     elidedRange,
     nextPage,
-    onPaginate,
     prevPage
   } = props
+  const [queryParams, setQueryParams] = useSearchParams()
+  const currPage = parseInt(queryParams.get('page')) || 1
 
   const getKeyStr = (num, idx) => {
     return `page-${typeof num !== 'number' ? `ellip-${idx}` : num}`
```

Comment:
I know this is not new, but still: whaat?

Reply:
now that you pointed it out: yes too hard to read. will try to make it more readable.